pkg/kubelet/checkpoint/checkpoint.go

```diff
@@ -103,7 +103,11 @@ func LoadPods(cpm checkpointmanager.CheckpointManager) ([]*v1.Pod, error) {
 			klog.Errorf("Failed to retrieve checkpoint for pod %q: %v", key, err)
 			continue
 		}
-		pods = append(pods, checkpoint.GetPod())
+                // clear NodeSelector when restoring from checkpoint otherwise pod cannot be restarted
```

Comment:
I am not convinced by this change. 

When kubelet does not has certain label applied, that said, I will expect the pod restored failed, no?

Reply:
If the pod is checkpointed, it means it met the nodeSelector at the time. So it shouldn't need that check again on checkpoint bringup.

Reply:
Yes that’s right. Until the kubelet connects to the api-server no labels are present. With no labels the nodeSelector on the checkpointed pods can never be satisfied.

Reply:
In principle, the kubelet needs to continue running pods that were previously admitted even if it loses connectivity with the API server. So instead of losing the node selector information, can you instead update the admission logic to ignore node selectors?

Reply:
I don't think kubelet ever reevaluates nodeselectors on running pods during the time its disconnected from the api server. So I don't think it buys you anything moving the logic there? You'd have to probably carry the info that it is a "checkpoined pod" all the way to the admission logic somehow, which probably would involve a pod api flag to for admission logic too? Is that worth it?

Reply:
@resouer @vishh Do you have more concerns about this change?